android/build.gradle

```diff
@@ -5,29 +5,19 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.2.1'
+        classpath 'com.android.tools.build:gradle:4.0.2'
     }
 }
 
 apply plugin: 'com.android.library'
-apply plugin: 'maven'
-
-def safeExtGet(prop, fallback) {
-    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
-}
-
-def DEFAULT_COMPILE_SDK_VERSION = 28
-def DEFAULT_BUILD_TOOLS_VERSION = "28.0.2"
-def DEFAULT_MIN_SDK_VERSION = 16
-def DEFAULT_TARGET_SDK_VERSION = 27
 
 android {
-  compileSdkVersion safeExtGet('compileSdkVersion', DEFAULT_COMPILE_SDK_VERSION)
-  buildToolsVersion safeExtGet('buildToolsVersion', DEFAULT_BUILD_TOOLS_VERSION)
+  compileSdkVersion 29
+  buildToolsVersion "29.0.2"
 
   defaultConfig {
-    minSdkVersion safeExtGet('minSdkVersion', DEFAULT_MIN_SDK_VERSION)
-    targetSdkVersion safeExtGet('targetSdkVersion', DEFAULT_TARGET_SDK_VERSION)
+    minSdkVersion 21
+    targetSdkVersion 29
```

Comment:
Are these changes necessary? Last I recall it is recommended to inherit the sdk versions from the parent application.

Reply:
This is a function of how things are packaged. There is no concept of inheriting parent application settings in libraries on Android. In that model, a library would have to know about how every project will use it. Typically, Android libraries are agnostic to how they're used and parent applications adjust to the libraries they decide to pull in. A library will declare which SDK levels it supports and targets and a parent can work around any limitations that creates.

From an Android perspective, I would suggest that the library make clear what APIs it can and cannot support.

Reply:
given my ignorance of android libraries, this is fine